azurelinuxagent/common/protocol/wire.py

```diff
@@ -1284,8 +1287,9 @@ def get_host_plugin(self):
         return self.host_plugin
 
     def has_artifacts_profile_blob(self):
-        return self.ext_conf and not \
-            textutil.is_str_none_or_whitespace(self.ext_conf.artifacts_profile_blob)
+        ext_conf = self.get_ext_conf()
```

Comment:
postpone until we remove the file cache?

Reply:
Done